engine/equationDisplay.cc

```diff
@@ -556,6 +556,54 @@ namespace MathDAG
     if (!arguments.empty() && !arguments[0].empty() && arguments[0][0])
       {parenthesise(surf, [&](Surface& surf){arguments[0][0]->render(surf);});}
   }
+  
+  template <>
+  void OperationDAG<OperationType::percent>::render(Surface& s) const
+  {
+      double xx,yy;
+      cairo_get_current_point(s.cairo(),&xx,&yy);
+ 
+      RecordingSurface r;
+      arguments[0][0]->render(r);
+
+      Pango pango(s.cairo());
+      double oldFs=pango.getFontSize();
+      pango.setFontSize(r.height());
+      cairo_rel_move_to(s.cairo(),0,-(r.height()-oldFs));
+      pango.show();
+      cairo_rel_move_to(s.cairo(),0,(r.height()-oldFs));
+      cairo_rel_move_to(s.cairo(),pango.width(),0);
+      arguments[0][0]->render(s);
+      xx+=pango.width()+r.width();
+      pango.setMarkup("%");
+      cairo_move_to(s.cairo(),xx,yy-r.height()+oldFs);
+      pango.show();
+      cairo_move_to(s.cairo(),xx+pango.width(),yy);
+  }
+  
+  template <>
+  void OperationDAG<OperationType::fact>::render(Surface& s) const
+  {
+      double xx,yy;
+      cairo_get_current_point(s.cairo(),&xx,&yy);
+ 
+      RecordingSurface r;
+      arguments[0][0]->render(r);
+
+      Pango pango(s.cairo());
+      double oldFs=pango.getFontSize();
+      pango.setFontSize(r.height());
+      cairo_rel_move_to(s.cairo(),0,-(r.height()-oldFs));
+      pango.show();
+      cairo_rel_move_to(s.cairo(),0,(r.height()-oldFs));
+      cairo_rel_move_to(s.cairo(),pango.width(),0);
+      arguments[0][0]->render(s);
+      xx+=pango.width()+r.width();
+      pango.setMarkup("!");
```

Comment:
Wouldn't you need to scale the ! symbol by the size of the text to the left?

Reply:
That's true, thanks. I will fix that now.

Reply:
Do you mean I should use recording surface s instead of r for the size of the ! and % symbols?